src/ARC1-limitedApproval.lua

```diff
@@ -103,10 +104,12 @@ function transferFromLtd(from, to, amount, ...)
   assert(_allowance[pair], "not approved")
   assert(_allowance[pair] >= amount, "insufficient allowance")
 
-  _transfer(from, to, amount, ...)
+  re = _transfer(from, to, amount, ...)
   _allowance[pair] = _allowance[pair] - amount
```

Comment:
calling an external contract before changing internal state can lead to re-entrancy attack. it must update the allowance before the transfer